homeassistant/helpers/trigger.py

```diff
@@ -409,3 +514,108 @@ def remove_triggers() -> None:
+            ),
+        )
+    except FileNotFoundError:
+        _LOGGER.warning(
+            "Unable to find triggers.yaml for the %s integration", integration.domain
+        )
+        return {}
+    except (HomeAssistantError, vol.Invalid) as ex:
+        _LOGGER.warning(
+            "Unable to parse triggers.yaml for the %s integration: %s",
+            integration.domain,
+            ex,
+        )
+        return {}
+
+
+def _load_triggers_files(
+    hass: HomeAssistant, integrations: Iterable[Integration]
+) -> dict[str, JSON_TYPE]:
+    """Load trigger files for multiple integrations."""
+    return {
+        integration.domain: _load_triggers_file(hass, integration)
+        for integration in integrations
+    }
+
+
+async def async_get_all_descriptions(
+    hass: HomeAssistant,
+) -> dict[str, dict[str, Any]]:
+    """Return descriptions (i.e. user documentation) for all triggers."""
+    descriptions_cache = hass.data[TRIGGER_DESCRIPTION_CACHE]
+
+    triggers = hass.data[TRIGGERS]
+    # See if there are new triggers not seen before.
+    # Any trigger that we saw before already has an entry in description_cache.
+    all_triggers = set(triggers)
+    # If we have a complete cache, check if it is still valid
+    if all_cache := hass.data.get(ALL_TRIGGER_DESCRIPTIONS_CACHE):
+        previous_all_triggers = set(all_cache)
+        # If the triggers are the same, we can return the cache
+        if previous_all_triggers == all_triggers:
+            return all_cache
+
+    # Files we loaded for missing descriptions
+    loaded: dict[str, JSON_TYPE] = {}
+    # We try to avoid making a copy in the event the cache is good,
+    # but now we must make a copy in case new triggers get added
+    # while we are loading the missing ones so we do not
+    # add the new ones to the cache without their descriptions
+    triggers = triggers.copy()
+
+    if missing_triggers := all_triggers.difference(descriptions_cache):
+        domains_with_missing_triggers = {
+            triggers[missing_trigger] for missing_trigger in missing_triggers
+        }
+        ints_or_excs = await async_get_integrations(hass, domains_with_missing_triggers)
+        integrations: list[Integration] = []
+        for domain, int_or_exc in ints_or_excs.items():
+            if type(int_or_exc) is Integration and int_or_exc.has_triggers:
+                integrations.append(int_or_exc)
+                continue
+            if TYPE_CHECKING:
+                assert isinstance(int_or_exc, Exception)
+            _LOGGER.debug(
+                "Failed to load triggers.yaml for integration: %s",
+                domain,
+                exc_info=int_or_exc,
+            )
+
+        if integrations:
+            loaded = await hass.async_add_executor_job(
+                _load_triggers_files, hass, integrations
+            )
+
+    # Build response
+    descriptions: dict[str, dict[str, Any]] = {}
```

Comment:
why do we need this new var? can't we just use `descriptions_cache` instead?

Reply:
You're right, it was not needed. Removed.